src/webgpu/shader/validation/decl/context_dependent_resolution.spec.ts

```diff
@@ -0,0 +1,308 @@
+export const description = `
+Tests that context dependent names do not participate in name resolution.
+That is, a declaration named the same as a context dependent name will not interfere.
+
+Context-dependent names:
+ * Attribute names
+ * Built-in value names
+ * Diagnostic severity control
+ * Diagnostic triggering rules
+ * Enable extensions
+ * Language extensions
+ * Swizzles
+ * Interpolation type
+ * Interpolation sampling
+`;
+
+import { makeTestGroup } from '../../../../common/framework/test_group.js';
+import { keysOf } from '../../../../common/util/data_tables.js';
+import { ShaderValidationTest } from '../shader_validation_test.js';
+
+export const g = makeTestGroup(ShaderValidationTest);
+
+const kAttributeCases = {
+  align: `struct S { @align(16) x : u32 }`,
+  binding: `@group(0) @binding(0) var s : sampler;`,
+  builtin: `@vertex fn main() -> @builtin(position) vec4f { return vec4f(); }`,
+  // const is not writable
+  // diagnostic is a keyword
+  group: `@group(0) @binding(0) var s : sampler;`,
+  id: `@id(1) override x : i32;`,
+  interpolate: `@fragment fn main(@location(0) @interpolate(flat) x : i32) { }`,
+  invariant: `@fragment fn main(@builtin(position) @invariant pos : vec4f) { }`,
+  location: `@fragment fn main(@location(0) x : f32) { }`,
+  must_use: `@must_use fn foo() -> u32 { return 0; }`,
+  size: `struct S { @size(4) x : u32 }`,
+  workgroup_size: `@compute @workgroup_size(1) fn main() { }`,
+};
+
+g.test('attribute_names')
+  .desc('Tests attribute names do not use name resolution')
+  .params(u =>
+    u
+      .combine('case', keysOf(kAttributeCases))
+      .beginSubcases()
+      .combine('decl', ['override', 'const', 'var<private>'] as const)
+  )
+  .fn(t => {
+    const code = `
+    ${t.params.decl} ${t.params.case} : u32 = 0;
+    ${kAttributeCases[t.params.case]}
+    `;
```

Comment:
I suggest also using the declared variable in a function. e.g. append
```
fn use_var() -> u32 { return ${t.params.decl}; }
```
This proves that the name isn't shadowed by the attribute name at all either.